tools/src/main/java/org/apache/kafka/tools/consumer/group/ConsumerGroupCommand.java

```diff
@@ -414,20 +438,27 @@ private void printMembers(Map<String, Entry<Optional<GroupState>, Optional<Colle
                             maxHostLen = Math.max(maxHostLen, memberAssignment.host.length());
                             maxClientIdLen = Math.max(maxClientIdLen, memberAssignment.clientId.length());
                             includeGroupInstanceId = includeGroupInstanceId || !memberAssignment.groupInstanceId.isEmpty();
+                            String currentAssignment = memberAssignment.assignment.isEmpty() ?
+                                MISSING_COLUMN_VALUE : getAssignmentString(memberAssignment.assignment);
+                            String targetAssignment = memberAssignment.targetAssignment.isEmpty() ?
+                                MISSING_COLUMN_VALUE : getAssignmentString(memberAssignment.targetAssignment);
+                            maxCurrentAssignment = Math.max(maxCurrentAssignment, currentAssignment.length());
+                            maxTargetAssignment = Math.max(maxTargetAssignment, targetAssignment.length());
                         }
                     }
                 }
 
                 String format0 = "%" + -maxGroupLen + "s %" + -maxConsumerIdLen + "s %" + -maxGroupInstanceIdLen + "s %" + -maxHostLen + "s %" + -maxClientIdLen + "s %-15s ";
                 String format1 = "%" + -maxGroupLen + "s %" + -maxConsumerIdLen + "s %" + -maxHostLen + "s %" + -maxClientIdLen + "s %-15s ";
+                String verboseFormat = "%-15s %" + -maxCurrentAssignment + "s %-15s %" + -maxTargetAssignment + "s %s";
 
                 if (includeGroupInstanceId) {
                     System.out.printf("\n" + format0, "GROUP", "CONSUMER-ID", "GROUP-INSTANCE-ID", "HOST", "CLIENT-ID", "#PARTITIONS");
                 } else {
                     System.out.printf("\n" + format1, "GROUP", "CONSUMER-ID", "HOST", "CLIENT-ID", "#PARTITIONS");
                 }
                 if (verbose)
-                    System.out.printf("%s", "ASSIGNMENT");
+                    System.out.printf(verboseFormat, "CURRENT-EPOCH", "CURRENT-ASSIGNMENT", "TARGET-EPOCH", "TARGET-ASSIGNMENT", "UPGRADED");
```

Comment:
Didn't we say that we would display `UPGRADED` only when there is non-upgraded members in the group?